src/main/groovy/life/qbic/datamodel/dtos/business/ProductId.groovy

```diff
@@ -124,4 +124,11 @@ class ProductId {
     String toString() {
         return "${type}_${uniqueId}"
     }
+
+    @Override
+    int compareTo(ProductId productId2) {
+        //if the types are equal compare by uniqueId
```

Comment:
I would suggest using the equals method here to make sure it behaves as expected.
```suggestion
        //if the types are equal compare by uniqueId
        if(this.equals(productId2)) return 0
```

Reply:
done :)

Reply:
Sorry but that is not what i meant :) 
The operator in the type comparison is fine. What I meant is, that the productId compareTo method needs to be in sync with the equals method of ProductId